### PR TITLE
Change to accept regex without /

### PR DESCRIPTION
**Pull request by vitormattos**
The parser of https://stoplight.io/ don't accept / on regex pattern.

To accept pattern with and without / I created a method to prepare the pattern before use.

### CODE REVIEW

**Comment by froschdesign** [comment]
> The parser of https://stoplight.io/ don't accept / on regex pattern.

It is irrelevant who accepts something or not. Only the specification is important: https://swagger.io/docs/specification/data-models/data-types/#pattern 😃

File: src/Base/Body.php [hunk 1]
```diff
@@ -111,13 +111,22 @@ protected function matchString($name, $schemaArray, $body, $type)
 
     private function checkPattern($name, $body, $pattern)
     {
+        $pattern = $this->preparePattern($pattern);
```
Comment (byjg) [inline src/Base/Body.php:114]: You can just do:

```
$pattern = "/" . rtrim(ltrim($pattern, "/"), "/") . "/"
```

**Comment by byjg** [comment]
You're right. The OpenAPI specification is the only one we should base on. 

Double-checking the link you sent with the specification makes sense for this PR since the initial and final `/` are not required for the RegEx ECMA 262 but just for the PHP `preg_match`.

**Review by byjg — CHANGES_REQUESTED** [review]

**Review by byjg — APPROVED** [review]

**Comment by byjg** [comment]
Thanks. Version 3.1.4 published.

**Pull request merged**